### ISSUE TITLE
Usage of smart apostrophe not flagged by Vale.Spelling

### ISSUE THREAD

**Issue by dagardner-nv** [jdkato/txtlint#1003]
### Check for existing issues

- [x] Completed

### Environment

* Debian Linux 12.10
* Conda
* 3.11.2 & 3.9.5

### Describe the bug / provide steps to reproduce it

Vale.Spelling will correctly flag the word `tooklkit's` but not `toolkit’s`.

Repro (from https://github.com/dagardner-nv/vale-apostrophe-test)
`README.md`:
```
# vale-apostrophe-test

## Issue
Vale will flag inappropriate apostrophes in the following sentence:
* The toolkit's plugin mechanism is extensible.

But not using the smart apostrophe:
* The toolkit’s plugin mechanism is extensible.

## Versions
* vale version 3.9.5
* vale version 3.11.2
```

`.vale.ini`:
```
MinAlertLevel = error

# Configs for markdown
[*{.md}]

BasedOnStyles = Vale
```

Result:
```
$ vale ./README.md 

 README.md
 5:7  error  Did you really mean             Vale.Spelling 
             'toolkit's'?                                  

✖ 1 error, 0 warnings and 0 suggestions in 1 file.
```

I reviewed issues #206 and #591 however I believe this issue is different as I'm not using asciidoc.
